bin/starter-pack/_main.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Plugin Name: {{extension_name}}
+ *
+ * @package WC_Admin
+ */
+
+/**
+ * Register the JS.
+ */
+function add_extension_register_script() {
+
+	if ( ! class_exists( 'Automattic\WooCommerce\Admin\Loader' ) || ! \Automattic\WooCommerce\Admin\Loader::is_admin_page() ) {
+		return;
+	}
+
+	wp_register_script(
+		'{{extension_slug}}',
+		plugins_url( '/build/index.js', __FILE__ ),
+		array(
+			'wp-hooks',
+			'wp-element',
+			'wp-i18n',
+			'wc-components',
+		),
+		filemtime( dirname( __FILE__ ) . '/build/index.js' ),
```

Comment:
If you are using the latest `wp-scripts`, these values should be provided automatically by the `*.asset.php` files automatically output by the [Dependency Extraction Webpack Plugin](https://github.com/WordPress/gutenberg/tree/master/packages/dependency-extraction-webpack-plugin) that is included with `@wordpress/scripts`.  You can see this at work in the wc blocks project.

Reply:
So I can provide an empty array here? What about `wc-components`?

Reply:
no you don't provide an empty array, you provide dependencies that are prepped for you automatically by the webpack plugin:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/a1717cbbe811466fefac04b141be60128e56c356/src/Assets.php#L155-L172

For wc scripts, you'd need to make sure they are registered with the webpack plugin:

Here: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/a1717cbbe811466fefac04b141be60128e56c356/bin/webpack-helpers.js#L24-L45

And: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/a1717cbbe811466fefac04b141be60128e56c356/bin/webpack-helpers.js#L272-L277

Reply:
Thanks for the explanation, I added the Dependency Extraction Webpack Plugin in https://github.com/woocommerce/woocommerce-admin/pull/3403/commits/7262f3d164c291d3bc87aa3179fb806317968ef9

You can test it out by placing something like this in the `index.js` file of the extension and watching webpack create the dependencies automatically.

```js
import { Button } from '@wordpress/components';
import { Chart } from '@woocommerce/components';

console.log( Button );
console.log( Chart );
```